client/test/io.js

```diff
@@ -0,0 +1,11 @@
+/**
+  * Mock class for io
+  */
+
+module.exports = () => ({
+  socket: {
+    postAsync: function () {
+      return Promise.resolve().then(() => ({}));
```

Comment:
Pretty sure you commented this in the wrong place, but yeah, I can add a test for that.